### ISSUE TITLE
Document HTML5 output dialect 

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#896]
CC-ing @brucemiller and @kohlhase here.

This issue is more of a question/discussion starter. It is related partially to #835, but is mainly due to a revival of our KWARC effort to bundle together and make available an HTML5 representation of the successful subset of arXiv under latexml.

Namely, it would help to have a level of documentation comparable to the [main schema](http://dlmf.nist.gov/LaTeXML/manual/schema/) manual, that specifically zeroes in on the HTML5 output. We have had this discussion internally at KWARC several times over the years, and based on our experience it turned out that working on a dialect of HTML is a much more adoptable approach for the wider community, than working on LaTeXML's custom XML schema. The main goal I have in mind right now is that when a newcomer contributor obtains the HTML5 arXMLiv corpus, they have a convenient way to gloss the different types of markup and supported constructs. It can be a very powerful starter for designing linguistic analysis tools, search engine indexers, etc.

If we agree having this level of documentation would be useful, another question is whether it would belong in the latexml manual, or should be hosted externally -- i'm open to both, although the manual feels like the natural host.

Happy to contribute to that myself, and it would also be a good time to see if there are some schema.org synergies that could be discussed in relation to latexml's (optional) RDFa output.

**Comment by kohlhase** [comment]
I think a documentation of the HTML5 is extremely useful and would indeed be timely. 

If I understand correctly, then this manual would mostly be a documentation of the generated RDFa and CSS classes, and what they mean correct? And as such it would be much less a "schema manual", but an "annotation lexicon", so I am wondering whether the schema.org synergies will be so important.

**Comment by kohlhase** [comment]
I think the LaTeXML manual would be the right place to put such a documentation with a machine-readable copy of the RDFa part on schema.org for outreach.

**Comment by dginev** [comment]
> mostly be a documentation of the generated RDFa and CSS classes, and what they mean correct?

Actually, I think that may be an oversimplification. But I see and agree with the point that we do not want to rewrite the HTML5 spec as part of documenting the latexml dialect. In this regard my comparison to the intermediate latexml XML is indeed misleading, as that is structured the way a traditional self-contained schema definition would be.

It is possible that the correct way to structure the html docs is to:
 * state that the dialect builds on and respects the main html5 specification
 * proceed to provide a map from "authoring constructs" to the specific html5 produced by latexml. 
 * a lot of these mappings happen in bindings, so there is a question whether we want to organize the documentation in a latex-centric way, namespaced per package, rather than a simple enumaration of classes and attributes. 
 * In fact both directions of access may be reasonable, example questions for a new latexml student:
   - I wonder what `\begin{theorem}` produces in html5?
   - I wonder what the `h6` tag is used for by latexml?
 
Back to the original point, there are non-trivial markup decisions to document for html5 that also include tags. Some of the examples I have seen recently while working in Authorea: 
 * The use of heading tags is not fixed to the latex heading level. For example a `\section` would be a `h1` if that is the top heading of a document, but `h2` if it is within a chapter. In this case the class provides the definitive semantics.
 * At the same time, `amsthm` environments such as `\begin{theorem}` more often than not end up as `h6` tags, and have the actual semantics encoded in the class attribute's values again. It would be great to have a documented answer on how latexml uses the HTML5  `h` elements in general, to avoid having to reverse-engineer the full answer from the code.
 * One of the most complex topics is of course aligned display equations - I believe Bruce pretty much invented the dual use of HTML tables and MathML, there is a paper on the cross-referencing algorithm etc.
 * Another very interesting bit to document is the bibliography markup

So this may be a little complicated to structure, and we may even need separate documents... Or instead one very high level document explaining all conventions used, and then only covering "notable bindings" as examples of using the conventions? Just throwing ideas and examples out there.

I have also now found the [css classes](http://dlmf.nist.gov/LaTeXML/manual/cssclasses/) section of the manual, so likely extending over that with a high level document could get us a long way. So this may end up closer to the [scholarly html](https://w3c.github.io/scholarly-html/) effort than to the intermediate schema.

**Comment by kohlhase** [comment]
> So this may end up closer to the scholarly html effort than to the intermediate schema.

yes, that was something I was thinking about as well. And another "client" of the documentation effort.

**Comment by bfirsh** [comment]
Happy to help contribute to this when there is something to add to!

Something I would also find useful is a comprehensive set of example latex documents with their corresponding HTML output. In some ways, I would find this more useful than documentation, in the same way that example code is often more useful than the documentation.

This could double as a test suite. I'll open a separate issue about that to avoid taking this off topic. :)

**Comment by brucemiller** [comment]
I would try (real hard) to claim that most of the mapping from latex to xml to html is Obvious; and even the css names are, once you've gotten the naming conventions.

I'd think that a more readable (& maintanable) document would list the places where the transformation is less obvious, or certain choices have to be made.  But then, I'm biased.

**Comment by kohlhase** [comment]
> But then, I'm biased.

you  certainly are. I think what Deyan is calling for (and rightfully so I think) is a document that explains the HTML dialect independently of LaTeXML, and to people who do not want to understand LaTeXML (or even LaTeX). 

I think that the LaTeXML dialect of HTML is one of the best bets of a Scientific Web Document Format  we have at the moment (since it comes from the well-thought-out LaTeX world) .

**Comment by dginev** [comment]
Closing due to lack of interest.
